tiledb/core.cc

```diff
@@ -1681,6 +1681,27 @@ py::str as_built_dump() {
   return res;
 }
 
+auto walk_callback = [](const char *path_ptr, tiledb_object_t obj,
+                        void *pyfunc) {
+  std::string my_path(path_ptr);
+  std::string my_objtype;
+  if (obj == TILEDB_GROUP) {
+    my_objtype = "group";
+  } else if (obj == TILEDB_ARRAY) {
+    my_objtype = "array";
+  } else {
+    return 0;
+  }
+  try {
+    py::function func =
+        py::reinterpret_borrow<py::function>((PyObject *)pyfunc);
```

Comment:
This is an unchecked cast, we should use `py::cast` here instead right? https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv4I0E18reinterpret_borrow1T6handle